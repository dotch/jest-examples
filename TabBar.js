import React from 'react';

const TabBar = ({items, selectedIndex}) => (
  <ul>
    {items.map((item, index) => (
      <li className={selectedIndex === index ? 'selected' : null} key={index}>
        {item}
      </li>
    ))}
  </ul>
);

export default TabBar;
