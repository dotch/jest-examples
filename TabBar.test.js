import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import {configure, shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import TabBar from './TabBar';

configure({adapter: new Adapter()});

describe('TabBar Component', () => {
  it('renders correctly', () => {
    const component = renderer
      .create(<TabBar items={['zebra', 'elephant']} selectedIndex={1} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('renders correctly (enzyme)', () => {
    const component = shallow(
      <TabBar items={['zebra', 'elephant']} selectedIndex={1} />
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
