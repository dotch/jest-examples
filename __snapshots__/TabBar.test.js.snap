// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBar Component renders correctly (enzyme) 1`] = `
<ul>
  <li
    className={null}
    key="0"
  >
    zebra
  </li>
  <li
    className="selected"
    key="1"
  >
    elephant
  </li>
</ul>
`;

exports[`TabBar Component renders correctly 1`] = `
<ul>
  <li
    className={null}
  >
    zebra
  </li>
  <li
    className="selected"
  >
    elephant
  </li>
</ul>
`;
